azuredata/src/main/java/com/azure/data/service/Response.kt

```diff
@@ -21,7 +19,13 @@ open class Response<T>(
         //  The json data returned by the server (if applicable)
         val jsonData: String? = null,
         // The result of response deserialization.
-        val result: Result<T>
+        val result: Result<T>,
+        // The resourceLocation, filled out when there could be more results
+        val resourceLocation: ResourceLocation? = null,
+        // The class of the document
+        val documentClass: Class<*>? = null,
+        // The DocumentClient used to acquire this Response
+        val documentClient: DocumentClient? = null
```

Comment:
Rather than passing in the `DocumentClient` here, let's use the AzureData 'static' calls since it will already have its doc client instance.  We can make the methods internal if it makes sense.  This should avoid the need to store the doc client in the response here and also support other, non Doc DB, data stores which should also support paging.

Reply:
The reason I chose to store the documentClient was if the user had created their own client, I thought it would be important to re-use that one instead of using the static one. Can the user ever build their own client and use it instead of using the static one in `AzureData`?